oaknational/Oak-Web-Application
# feat(curric): update unit visualiser data

## Description
- Updated `CurriculumUnitsQuery` with new data model from MV
- Updated unit visualiser tab to support new data
- Added new `TagFunctional` component
- Added tag with unit option count to unit cards with more than 1 option
- When multiple programmes are shown (as in Secondary Science) ordered threads alphabetically
- In other cases, ordered threads by programme thread order

## How to test

1. Go to [teachers/curriculum/english-primary/units](https://deploy-preview-1915--oak-web-application.netlify.thenational.academy/teachers/curriculum/english-primary/units)
2. Scroll to Year 2
4. You should see a tag for units with multiple options reading "{x} unit options"
5. You should threads in the sidebar ordered by the order specified in programme threads
6. Got to [teachers/curriculum/science-secondary/units](https://deploy-preview-1915--oak-web-application.netlify.thenational.academy/teachers/curriculum/science-secondary/units)
7. You should threads in the sidebar ordered alphabetically

## Screenshots

How it used to look (delete if n/a):
<img width="966" alt="Screenshot 2023-09-22 at 12 30 56" src="https://github.com/oaknational/Oak-Web-Application/assets/421186/24dd12e5-0530-4841-b902-acef896b0c44">
<img width="505" alt="Screenshot 2023-09-22 at 12 31 43" src="https://github.com/oaknational/Oak-Web-Application/assets/421186/ef9029b9-0844-4264-8a8d-dffeab903a50">

How it should now look:
<img width="966" alt="Screenshot 2023-09-22 at 12 30 27" src="https://github.com/oaknational/Oak-Web-Application/assets/421186/15187955-1bf3-4df8-874a-60fe53627f92">
<img width="505" alt="Screenshot 2023-09-22 at 12 31 48" src="https://github.com/oaknational/Oak-Web-Application/assets/421186/168c58d0-6361-4e8c-b0a0-35ed3a47dd80">

## Checklist
